crates/bitwarden-core/src/uniffi_support.rs

```diff
@@ -2,47 +2,22 @@
 
 use std::num::NonZeroU32;
 
-use bitwarden_crypto::{EncString, UnsignedSharedKey};
 use uuid::Uuid;
 
-use crate::UniffiCustomTypeConverter;
-
-uniffi::ffi_converter_forward!(NonZeroU32, bitwarden_crypto::UniFfiTag, crate::UniFfiTag);
-uniffi::ffi_converter_forward!(EncString, bitwarden_crypto::UniFfiTag, crate::UniFfiTag);
-uniffi::ffi_converter_forward!(
-    UnsignedSharedKey,
-    bitwarden_crypto::UniFfiTag,
-    crate::UniFfiTag
-);
+uniffi::use_remote_type!(bitwarden_crypto::NonZeroU32);
 
 type DateTime = chrono::DateTime<chrono::Utc>;
-uniffi::custom_type!(DateTime, std::time::SystemTime);
-
-impl UniffiCustomTypeConverter for chrono::DateTime<chrono::Utc> {
-    type Builtin = std::time::SystemTime;
-
-    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
-        Ok(Self::from(val))
-    }
-
-    fn from_custom(obj: Self) -> Self::Builtin {
-        obj.into()
-    }
-}
-
-uniffi::custom_type!(Uuid, String);
-
-impl UniffiCustomTypeConverter for Uuid {
-    type Builtin = String;
-
-    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
-        Uuid::parse_str(val.as_str()).map_err(|e| e.into())
-    }
-
-    fn from_custom(obj: Self) -> Self::Builtin {
-        obj.to_string()
-    }
-}
+uniffi::custom_type!(DateTime, std::time::SystemTime, {
+    remote,
+    try_lift: |val| Ok(Self::from(val)),
+    lower: |obj| obj.into(),
+});
```

Comment:
suggestion: trivial conversion using `From`/`TryFrom` traits is built-in into the macro from my understanding. I was able to simplify it like this:

```suggestion
uniffi::custom_type!(DateTime, std::time::SystemTime, { remote });
```

However I got blocked on the `use_remote_type` stuff so I'm not 100% sure if it actually worked

Reply:
Nice, checked the generated code and tested the ios and android apps, seems to be working the same